GitHubComments.ps1

```diff
@@ -139,33 +61,58 @@ function Get-GitHubRepositoryComment
         If not supplied here, the DefaultNoStatus configuration property value will be used.
 
     .EXAMPLE
-        Get-GitHubRepositoryComment -OwnerName Powershell -RepositoryName PowerShellForGitHub -Sort created -Direction asc -Since 2011-04-14T16:00:49Z
+        Get-GitHubComment-OwnerName Powershell -RepositoryName PowerShellForGitHub
 
         Get the comments for the PowerShell\PowerShellForGitHub project.
 #>
     [CmdletBinding(
         SupportsShouldProcess,
-        DefaultParametersetName='Elements')]
+        DefaultParametersetName='RepositoryElements')]
     [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "", Justification="Methods called within here make use of PSShouldProcess, and the switch is passed on to them inherently.")]
     param(
-        [Parameter(ParameterSetName='Elements')]
+        [Parameter(Mandatory, ParameterSetName='RepositoryElements')]
+        [Parameter(Mandatory, ParameterSetName='IssueElements')]
+        [Parameter(Mandatory, ParameterSetName='CommentElements')]
         [string] $OwnerName,
 
-        [Parameter(ParameterSetName='Elements')]
+        [Parameter(Mandatory, ParameterSetName='RepositoryElements')]
+        [Parameter(Mandatory, ParameterSetName='IssueElements')]
+        [Parameter(Mandatory, ParameterSetName='CommentElements')]
         [string] $RepositoryName,
 
-        [Parameter(
-            Mandatory,
-            ParameterSetName='Uri')]
+        [Parameter(Mandatory, ParameterSetName='RepositoryUri')]
+        [Parameter(Mandatory, ParameterSetName='IssueUri')]
+        [Parameter(Mandatory, ParameterSetName='CommentUri')]
         [string] $Uri,
 
+        [Parameter(Mandatory, ParameterSetName='CommentUri')]
+        [Parameter(Mandatory, ParameterSetName='CommentElements')]
+        [string] $CommentID,
+
+        [Parameter(Mandatory, ParameterSetName='IssueUri')]
+        [Parameter(Mandatory, ParameterSetName='IssueElements')]
+        [int] $IssueNumber,
+
+        [Parameter(ParameterSetName='RepositoryUri')]
+        [Parameter(ParameterSetName='RepositoryElements')]
+        [Parameter(ParameterSetName='IssueElements')]
+        [Parameter(ParameterSetName='IssueUri')]
+        [DateTime] $Since,
+
+        [Parameter(ParameterSetName='RepositoryUri')]
+        [Parameter(ParameterSetName='RepositoryElements')]
         [ValidateSet('created', 'updated')]
         [string] $Sort,
 
+        [Parameter(ParameterSetName='RepositoryUri')]
+        [Parameter(ParameterSetName='RepositoryElements')]
         [ValidateSet('asc', 'desc')]
         [string] $Direction,
 
-        [DateTime] $Since,
+        [ValidateSet('raw', 'text', 'html', 'full', '')]
+        [string] $MediaType,
+
+        [string] $MediaTypeVersion = 'VERSION',
```

Comment:
There are better ways to determine if the user specified a value or not.  I would recommend:
```powershell
[ValidateNotNullOrEmpty()]
[string] $MediaTypeVersion = ''
```
The validation specified _only_ happens if the caller uses this parameter and _does not_ happen for the default value you supplied. This means the caller is _forced_ to provide a valid, nonempty version string if they use this parameter and, in your code, a value of empty string means the caller did not supply a value
```powershell
Get-GitHubComment @args -MediaTypeVersion 'someVersion' # works
Get-GitHubComment @args -MediaTypeVersion '' # fails
Get-GitHubComment @args -MediaTypeVersion $null # fails
Get-GitHubComment @args # -MediaTypeVersion not specified, will be empty string inside the function's code.
```

Reply:
If this param is not supplied it should default to 'VERSION' since this is what is used in the docs when the latest version is requested. I am not too worried about validating if they pass null as invalid values will just default to the latest version being returned.

Reply:
Ok, that makes sense then.